actuators/actuators.go

```diff
@@ -71,6 +72,22 @@ func kubeconfigToSecret(clusterName, namespace string) (*v1.Secret, error) {
 		return nil, errors.WithStack(err)
 	}
 
+	allNodes, err := nodes.List(fmt.Sprintf("label=%s=%s", constants.ClusterLabelKey, clusterName))
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	// This is necessary so the management cluster in a container can talk to another container.
```

Comment:
This is the big hack.

The Management cluster, running the NodeRef controller needs to talk to the child clusters.

The management cluster and child cluster must communicate over DockerIP address/ports.

The load balancer listens on <docker_ip>:6443 and exposes a port on the host at some random open port. Any traffic directed to the nginx container will get sent over to all the control plane nodes in the cluster.

Since the NodeRef controller is running inside a container, it must reference the child cluster load balancer host by using the Docker IP address and port 6443, but us, running on the docker host, must use the localhost and random port the LB is exposing to our system.

Right now the secrets will work only for the node ref and in order for us to interact with the child clusters via kubeconfig we take the secret uploaded, rewrite the kube-apiserver-address to be 127.0.0.1:<randomport>. It's not perfect but it works to at least play with cluster-api v0.1.4

Reply:
I'd add this description to the code expanding the comments